netqasm/sdk/epr_socket.py

```diff
@@ -382,14 +651,106 @@ def create_context(
             )
             yield output, pair
         finally:
-            self.conn._builder._post_epr_context(
+            self.conn.builder._post_epr_context(
                 pre_commands=pre_commands,
                 number=number,
                 loop_register=loop_register,
                 ent_results_array=ent_results_array,
                 pair=pair,
             )
 
+    def recv_keep(
```

Comment:
next 3 methods almost identical. Alternative: give method to call and post_routine as an parameter to a generic method handling the 3 duplicate codes.

Reply:
In shape they are almost identical, but they each have a different function signature (i.e. different types), different docstrings, and different functions called inside. So I would rather keep them separate as they are.